examples/stable_diffusion_v2/conditions/openpose/body.py

```diff
@@ -0,0 +1,273 @@
+import math
+
+import cv2
+import matplotlib.pyplot as plt
+import numpy as np
+from scipy.ndimage import gaussian_filter
+
+import mindspore as ms
+
+from . import util
+from .model import bodypose_model
+
+
+class Body(object):
+    def __init__(self, model_path):
+        self.model = bodypose_model()
+        model_dict = ms.load_checkpoint(model_path)
+        param_not_load, ckpt_not_load = ms.load_param_into_net(self.model, model_dict)
+        self.model.set_train(False)
+
+    def __call__(self, oriImg):
+        # scale_search = [0.5, 1.0, 1.5, 2.0]
+        scale_search = [0.5]
+        boxsize = 368
+        stride = 8
+        padValue = 128
+        thre1 = 0.1
+        thre2 = 0.05
+        multiplier = [x * boxsize / oriImg.shape[0] for x in scale_search]
+        heatmap_avg = np.zeros((oriImg.shape[0], oriImg.shape[1], 19))
+        paf_avg = np.zeros((oriImg.shape[0], oriImg.shape[1], 38))
+
+        for m in range(len(multiplier)):
+            scale = multiplier[m]
+            imageToTest = cv2.resize(oriImg, (0, 0), fx=scale, fy=scale, interpolation=cv2.INTER_CUBIC)
+            imageToTest_padded, pad = util.padRightDownCorner(imageToTest, stride, padValue)
+            im = np.transpose(np.float32(imageToTest_padded[:, :, :, np.newaxis]), (3, 2, 0, 1)) / 256 - 0.5
+            im = np.ascontiguousarray(im)
+
+            data = ms.Tensor.from_numpy(im)
+            # data = data.permute([2, 0, 1]).unsqueeze(0).float()
+
+            # cong TODO: make sure below 1 line code runs without gradients update
+            Mconv7_stage6_L1, Mconv7_stage6_L2 = self.model(data)
```

Comment:
maybe use `ops.stop_gradients` here to stop gradients backpropagtion

Reply:
got it!